core/drivers/clk/clk.c

```diff
@@ -14,23 +14,61 @@
 /* Global clock tree lock */
 static unsigned int clk_lock = SPINLOCK_UNLOCK;
 
+void clk_init_instance(struct clk *clk, const struct clk_ops *ops)
+{
+	assert(clk && is_clk_elt(clk) || is_clk_orphan(clk) && ops);
```

Comment:
assert is incorrect as you may intend to access a clk->ops that is not yet initialized in is_clk_elt() and is_clk_orphan.

Reply:
indeed something's wrong here. thanks.

Reply:
addressed in fixup commit.